sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -95,6 +106,31 @@ abstract class StatePartitionReaderBase(
       schema, "value").asInstanceOf[StructType]
   }
 
+  protected val getStoreUniqueId : Option[String] = {
+    val partitionStateUniqueIds =
+      partition.sourceOptions.operatorStateUniqueIds.map(_(partition.partition))
+    if (partition.sourceOptions.storeName == StateStoreId.DEFAULT_STORE_NAME) {
+      partitionStateUniqueIds.map(_.head)
+    } else {
+      val stateStoreCheckpointIds = SymmetricHashJoinStateManager.getStateStoreCheckpointIds(
+        partition.partition,
+        partition.sourceOptions.operatorStateUniqueIds,
+        useColumnFamiliesForJoins = false)
+
+      partition.sourceOptions.storeName match {
```

Comment:
You don't need to do this here. This can be done within the join call above and it will just return the id you need for the storeName, instead of returning the entire `stateStoreCheckpointIds`.